docs/topics/db/queries.txt

```diff
@@ -1134,6 +1134,18 @@ array:
     >>> Dog.objects.filter(data__owner__other_pets__0__name="Fishy")
     <QuerySet [<Dog: Rufus>]>
 
+If the key is a negative integer, it cannot be used in a filter keyword
+directly, but you can still use dictionary unpacking to use it in a query:
+
+.. code-block:: pycon
+
+    >>> Dog.objects.filter(**{"data__owner__other_pets__-1__name": "Fishy"})
+    <QuerySet [<Dog: Rufus>]>
+
+.. admonition:: MySQL and MariaDB
+
+    Using negative JSON array indices is not currently supported.
+
```

Comment:
Can you add a `.. versionchanged:: 6.0` note for this and a release note for 6.0?